components/org.wso2.identity.event.websubhub.publisher/src/main/java/org/wso2/identity/event/websubhub/publisher/config/OutboundAdapterConfigurationProvider.java

```diff
@@ -0,0 +1,99 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.identity.event.websubhub.publisher.config;
+
+import edu.umd.cs.findbugs.annotations.SuppressWarnings;
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.identity.event.common.publisher.exception.AdapterConfigurationException;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.channels.Channels;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.Optional;
+import java.util.Properties;
+
+import static java.util.Objects.isNull;
+import static java.util.Optional.empty;
+import static java.util.Optional.ofNullable;
+import static org.wso2.identity.event.websubhub.publisher.constant.WebSubHubAdapterConstants.CONFIG_FILE_NAME;
+
+/**
+ * Class to build the output adapter configurations.
+ */
+public class OutboundAdapterConfigurationProvider {
+
+    private final Properties adapterProperties;
+    private static OutboundAdapterConfigurationProvider instance;
+
+    private OutboundAdapterConfigurationProvider() throws AdapterConfigurationException {
+
+        adapterProperties = this.loadProperties();
+    }
+
+    public static OutboundAdapterConfigurationProvider getInstance() throws AdapterConfigurationException {
+
+        if (instance == null) {
+            instance = new OutboundAdapterConfigurationProvider();
+        }
+        return instance;
+    }
+
+    @SuppressWarnings("PATH_TRAVERSAL_IN")
+    private Properties loadProperties() throws AdapterConfigurationException {
+
+        Properties properties = new Properties();
+
+        Path path = Paths.get(IdentityUtil.getIdentityConfigDirPath(), CONFIG_FILE_NAME);
+
+        if (Files.notExists(path)) {
+            throw new AdapterConfigurationException(CONFIG_FILE_NAME + " configuration file doesn't exist.");
+        }
+
+        FileChannel channel;
+        try {
+            channel = FileChannel.open(path, StandardOpenOption.READ);
+
+            try (InputStream inputStream = Channels.newInputStream(channel)) {
+                properties.load(inputStream);
+            }
+        } catch (IOException e) {
+            throw new AdapterConfigurationException("Error while retrieving the configuration file.", e);
+        }
+
+        return properties;
+    }
+
+    /**
+     * Returns the {@link Optional} value of the property provided.
+     *
+     * @param propertyName name of the config property.
+     * @return Optional value of the property provided.
+     */
+    public Optional<String> getProperty(String propertyName) {
```

Comment:
Returning "Optional" is not a good practice due to high cost of memory. Do we really need this ? Why can't we simply return the String here ?

Reply:
```
configurationProvider.getProperty(HTTP_CONNECTION_TIMEOUT).map(Integer::parseInt).orElse(
                        WebSubHubAdapterConstants.DEFAULT_HTTP_CONNECTION_TIMEOUT);
```

Like in here, there are default values to be return when property is not set. Using optional makes the nullability explicit, enhances readability and clarity of intent.

Reply:
Fixed with https://github.com/wso2-extensions/identity-event-publishers/pull/1/commits/eb4cae044fe65111cb72917fd78af0619d056a02